src/App.js

```diff
@@ -90,20 +92,34 @@ class App extends Component {
     return (
       <>
         <Navbar user={this.state.user} logOutUser={this.logOutUser} />
-        <Router>
-          <Home path="/" user={this.state.user} />
-          <Login path="login" />
-          <Register path="register" registerUser={this.registerUser} />
-          <Feed path="recipes" />
-          <Recipe path="recipes/:recipeId" />
-          <Profile path="/profile/:userId" />
-          <Form user={this.state.user} path="/add-recipe" />
-          <DatabaseTests path="/database-tests" />
-          <HTTP_404 default />
-        </Router>
+        <UserContext.Provider value={this.state}>
+          <Router>
+            <Home path="/" user={this.state.user} />
+            <Login path="login" />
+            <Register path="register" registerUser={this.registerUser} />
+            <Feed path="recipes" />
+            <Recipe path="recipes/:recipeId" />
+            <PrivateRoute as={Profile} path="/profile/:userId" />
+            <PrivateRoute as={Form} user={this.state.user} path="/add-recipe" />
+            <DatabaseTests path="/database-tests" />
+            <HTTP_404 default />
+          </Router>
+        </UserContext.Provider>
       </>
     );
   }
 }
 
+class PrivateRoute extends React.Component {
+  static contextType = UserContext;
+
+  render() {
+    let { as: Comp, ...props } = this.props;
+    if (this.context.user) {
+      return <Comp {...props} />;
+    }
+    return <HTTP_404 />;
```

Comment:
Was hältst du davon, wenn wir der 404-Komponente noch den `message`-prop mitgeben? Dann weiß man auch gleich wieso man die 404 Seite bekommt :) 
```suggestion
    return <HTTP_404 message="Please login first" />;
```